parsers/dortmund.py

```diff
@@ -0,0 +1,143 @@
+    128: 'Schnelle Theke',
+    130: 'Sonstige',
+    131: 'Low Carb Teller',
+    135: 'Pastatheke Pastaauswahl',
+    136: 'Pastatheke Toppings',
+    137: 'Pastatheke Saucen',
+}
+
+def getWeekdays(day):
+    currentWeekday = day.weekday()
+    week = []
+    wstart = -currentWeekday
+    wend = 7 - currentWeekday
+    for d in range(wstart, wend):
+        nday = day + datetime.timedelta(days=d)
+        week.append(nday.isoformat())
+    return week
+
+def parse_url(url, today=False):
+    canteen = LazyBuilder()
+
+    canteen.extra_regex = re.compile('\((?P<extra>[0-9a-zA-Z]{1,3}'
+                                     '(?:,[0-9a-zA-Z]{1,3})*)\)', re.UNICODE)
+
+    legend_url = 'https://www.stwdo.de/mensa-co/allgemein/zusatzstoffe/'
+    canteen.legend = parse_legend(legend_url)
+
+    day = datetime.date.today()
+    week = getWeekdays(day)
+
+    for wDay in week:
+        py = {'tx_pamensa_mensa[date]' : wDay}
+        payload = urlencode(py).encode('ascii')
+        data = rq.urlopen(url, payload).read().decode('utf-8')
+        soup = BeautifulSoup(data, 'html.parser')
+        parse_day(canteen, soup, wDay)
+
+    return canteen.toXMLFeed()
+
+def parse_legend(url):
+    data = rq.urlopen(url).read().decode('utf-8')
+    soup = BeautifulSoup(data, 'html.parser')
+
+    table = soup.find('table', { 'class' : 'ce-table' })
+    tbody = soup.find('tbody')
+
+    legend = {}
+
+    for tr in tbody.find_all('tr'):
+        td = tr.find_all('td')
+        key = td[0].text.strip(' ')
+        legend[key] = td[1].text
+
+    return legend
+
+def define_category(item, img):
+    if img is not None and len(img['title']) > 0:
+        return img['title']
+    else:
+        catNumber = int(re.findall('[0-9]{3}', item['class'][2])[0])
+        fallback = categories.get(130)
+        return categories.get(catNumber, fallback)
+
+def getAndFormatPrice(price):
+    price = re.search('(\d+),(\d{2})', price)
+    if price is not None:
+        formatted = int(price.group(1) + price.group(2))
+        return formatted
+    else:
+        return '-'
+
+def parse_day(canteen, soup, wdate):
+    mealsBody = soup.find('div', { 'class' : 'meals-body' })
+
+    for meal in mealsBody.find_all('div', { 'class' : 'meal-item' }):
+        for item in meal.find_all('div', { 'class' : 'item' }):
+            if 'category' in item['class']:
+                img = item.find('img')
+                category = define_category(item, img)
+            elif 'description' in item['class']:
+                description = item.text.strip()
+            elif 'supplies'in item['class']:
+                supplies = []
+                for supply in item.find_all('img'):
+                    if supply['title']:
+                        supplies.append(supply['title'])
+            elif 'price'in item['class']:
+                price = item.text
+                if 'student' in item['class']:
+                    student_price = getAndFormatPrice(price)
```

Comment:
I saw in your commits, that you removed the `getAndFormatPrice` function and then added it back. I just tested it myself and the `convertPrice` function in feed.py seems to properly extract all prices. Were there some problems with just passing the string directly?

Reply:
Yeah, you are right I removed the function, but then I realised again why I put it in in the first place. There are some meals (I guess only side dishes), where there is no price given. The standard parsing of the prices breaks at that point, so I implemented my method to handle empty prices.

Reply:
You are right. Seems like we should make `convertPrice` a little bit more flexible. :sweat_smile: Do you remember an example where there were no prices given?